aes-gcm/src/xaes.rs

```diff
@@ -0,0 +1,127 @@
+use core::ops::{Div, Mul};
+
+use aead::{array::Array, AeadCore, AeadInPlace, Error, Key, KeyInit, KeySizeUser};
+use cipher::{consts::U2, BlockCipherEncrypt, BlockSizeUser};
+
+use crate::{Aes256, Aes256Gcm, Nonce, Tag};
+
+/// XAES-256-GCM
+#[derive(Clone)]
+pub struct XaesGcm256 {
+    aes: Aes256,
+    k1: Block,
+}
+
+type KeySize = <Aes256Gcm as KeySizeUser>::KeySize;
+type NonceSize = <<Aes256Gcm as AeadCore>::NonceSize as Mul<U2>>::Output;
+type TagSize = <Aes256Gcm as AeadCore>::TagSize;
+type CiphertextOverhead = <Aes256Gcm as AeadCore>::CiphertextOverhead;
+type Block = Array<u8, <Aes256 as BlockSizeUser>::BlockSize>;
+
+/// Maximum length of plaintext.
+pub const P_MAX: u64 = 1 << 36;
+
+/// Maximum length of associated data.
+// pub const A_MAX: u64 = 1 << 61;
```

Comment:
probably forgot to delete ?

Reply:
I left the code here to inspire this comment. :)

According to the C2SP spec, the associated data max size for XAES-256-GCM is 2EiB = 2^61 bytes. However the max size for AD for AES-GCM is 2^64 and yet it's been restricted to 2^36 by this crate, so specifying the "true" AD would have no affect. I didn't explore why the AD is limited in this manner, however, so I left the code there in hopes that that would be clarified.